src/libs/ReportUtils.js

```diff
@@ -1437,6 +1483,27 @@ function getReportPreviewMessage(report, reportAction = {}) {
         return `approved ${formattedAmount}`;
     }
 
+    /* === Start - Check for Report Scan === */
+
+    if (shouldConsiderReceiptBeingScanned) {
+        const filteredIouReportActions = getSortedMoneyRequestActions(report.reportID);
+        const lastIouReportAction = _.first(filteredIouReportActions);
+        const lastIouReportActionTransaction = TransactionUtils.getLinkedTransaction(lastIouReportAction);
+
+        // get lastTransaction
+        // eslint-disable-next-line no-use-before-define
+        const transactionsWithReceipts = getSortedTransactionsWithReceipts(report.reportID);
+        const lastTransaction = _.first(transactionsWithReceipts);
+
+        const transactionIsLastReportAction = _.isEqual(lastIouReportActionTransaction, lastTransaction);
+
+        if (lastTransaction && transactionIsLastReportAction && TransactionUtils.isReceiptBeingScanned(lastTransaction)) {
```

Comment:
Yes I know that's being passed but I don't remember why I chose this approach. Now that I am not able to come up with a reason, we can definitely use the `reportAction` being passed to the funciton

Reply:
@allroundexperts Would you like to do the same here as well? 
![image](https://github.com/Expensify/App/assets/65986357/65d8f793-0bec-4381-bfb2-c5b88fd3dfa9)

Reply:
I mean using `reportAction` thats passed to the function doesn't seem to have any issue. I will proceed with what you mentioned. In my last comment, I said I don't remember why i used the approach that i did instead of directly using the `reportAction` this is passed.